src/Microsoft.Dnx.Tooling/Publish/PublishProject.cs

```diff
@@ -129,7 +129,18 @@ private bool EmitNupkg(PublishRoot root)
             if (root.Frameworks.Any())
             {
                 // Make sure we only emit the nupkgs for the specified frameworks
-                buildOptions.TargetFrameworks.AddRange(root.Frameworks);
+                // We need to pick actual project frameworks relevant to the publish
+                // but the project may not target exactly the right framework so we need
+                // to use a compatibility check.
+                foreach(var framework in root.Frameworks)
+                {
+                    var selectedFramework = project.GetCompatibleTargetFramework(framework.Key);
+                    if(selectedFramework == null)
+                    {
+                        root.Reports.WriteError($"Unable to build {project.Name}. It is not compatible with the requested target framework: {framework.Key}");
```

Comment:
You don't return here? Won't the following `selectedFramework.FrameworkName` cause `NullReferenceException`?